nessie/src/main/java/org/apache/iceberg/nessie/NessieUtil.java

```diff
@@ -102,6 +102,38 @@ private static String commitAuthor(Map<String, String> catalogOptions) {
         .orElseGet(() -> System.getProperty("user.name"));
   }
 
+  private static void checkAndUpdateGCProperties(
+      TableMetadata tableMetadata, Map<String, String> updatedProperties, String identifier) {
+    if (tableMetadata.propertyAsBoolean(
+        NessieTableOperations.NESSIE_GC_NO_WARNING_PROPERTY, false)) {
+      return;
+    }
+
+    // To prevent accidental deletion of files that are still referenced by other branches/tags,
+    // setting GC_ENABLED to 'false' is recommended, so that all Iceberg's gc operations like
+    // expire_snapshots, remove_orphan_files, drop_table with purge will fail with an error.
+    // `nessie-gc` CLI provides a reference-aware GC functionality for the expired/unreferenced
+    // files.
+    // Advanced users may still want to use the simpler Iceberg GC tools iff their Nessie Server
+    // contains only one branch (in which case the full Nessie history will be reflected in the
+    // Iceberg sequence of snapshots).
+    if (tableMetadata.propertyAsBoolean(
+            TableProperties.GC_ENABLED, TableProperties.GC_ENABLED_DEFAULT)
```

Comment:
does nessie really want to default to gc enabled here if the flag isn't set for some reason?

Reply:
We're just trying to do the same property lookup as Iceberg GC code would do.

Reply:
`tableMetadata.propertyAsBoolean()` would return `false` here, if the table-default setting is false, correct?

Reply:
This check basically does the same lookup as https://github.com/apache/iceberg/blob/513a2d803245726eb211cb6a0c14abcdb397e9f3/spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java#L352